tools/evtool/evtindex.hpp

```diff
@@ -114,5 +114,45 @@ class CStreamingBucketVisitor : public CInterfaceOf<IBucketVisitor>
     bool arrived{false};
 };
 
+class CIndexEventsOp : public CEventConsumingOp
+{
+public:
+    bool filterFileId(const char* fileId);
+    bool filterFilePath(const char* path);
+};
+
+// Extension of `TEventConsumingCommand` that provides a common base for most, if not all, commands
+// related to index file event analysis.
+//
+// `event_consuming_op_t` must implement the following extended interface:
+// - `bool filterFileId(const char* fileId)`: adds a file ID filter to the operation
+// - `bool filterFilePath(const char* path)`: adds a file path filter to the operation
+template <typename event_consuming_op_t>
```

Comment:
I am not a big fan of using template classes like this for a couple of reasons
i) the code is hard to follow
ii) The code gets replicated into every derived class, bumping up the generated code size.

An alternative is to add the functionality in the base class, with a flag that is checked to see if it is applicable.
Something to discuss the relative merits.